packages/docusaurus-theme-classic/src/theme/MDXComponents/index.tsx

```diff
@@ -33,40 +33,25 @@ const MDXComponents: MDXComponentsObject = {
     return <Head {...props}>{unwrappedChildren}</Head>;
   },
   code: (props) => {
-    const {children} = props;
-
-    // For retrocompatibility purposes (pretty rare use case)
-    // See https://github.com/facebook/docusaurus/pull/1584
-    if (isValidElement(children)) {
-      return children;
-    }
-
-    return !children.includes('\n') ? (
-      <code {...props} />
-    ) : (
-      <CodeBlock {...props} />
+    const shouldBeInline = React.Children.toArray(props.children).every(
+      (el) => typeof el === 'string' && !el.includes('\n'),
     );
-  },
-  a: (props) => <Link {...props} />,
-  pre: (props) => {
-    const {children} = props;
 
-    // See comment for `code` above
-    if (isValidElement(children) && isValidElement(children?.props?.children)) {
```

Comment:
wouldn't it be better to just improve this test?

Reply:
I simply don't think it's ever justified to unwrap the `pre` or `code` element... Users put it there for a reason and we have to make it render something they expect

Reply:
👍 agree, was just a random though